modules/renter/download.go

```diff
@@ -0,0 +1,150 @@
+package renter
+
+import (
+	"crypto/rand"
+	"errors"
+	"io"
+	"os"
+	"time"
+
+	"github.com/NebulousLabs/Sia/crypto"
+	"github.com/NebulousLabs/Sia/modules"
+)
+
+// A Download is a download that has been queued by the renter.
+type Download struct {
+	completed   bool
+	destination string
+	nickname    string
+
+	renter *Renter
+}
```

Comment:
yes. It's actually a pretty common pattern; for example, [Sha256.New](http://golang.org/src/crypto/sha256/sha256.go?s=1643:1663#L72) returns an unexported type which satisfies `hash.Hash` In fact, all of the module types (renter, host, etc.) could be unexported if their `New` function returned an interface instead of the concrete type.
...that might be a good idea, actually.